examples/gno.land/p/demo/stack/stack.gno

```diff
@@ -0,0 +1,43 @@
+package stack
+
+type Stack struct {
+	top    *node
+	length int
+}
+
+type node struct {
+	value interface{}
+	prev  *node
+}
+
+func New() *Stack {
+	return &Stack{nil, 0}
+}
+
+func (s *Stack) Len() int {
+	return s.length
+}
+
+func (s *Stack) Top() interface{} {
+	if s.length == 0 {
+		return nil
+	}
+	return s.top.value
+}
+
+func (s *Stack) Pop() interface{} {
+	if s.length == 0 {
+		return nil
+	}
+
+	node := s.top
+	s.top = node.prev
+	s.length -= 1
+	return node.value
+}
+
+func (s *Stack) Push(value interface{}) {
+	node := &node{value, s.top}
+	s.top = node
```

Comment:
Nitpick, but you can inline this:
`s.top = &node{value, s.top}`

Reply:
Here, I'm bit inclined to keep this style.